src/pocketdb/repositories/TransactionRepository.cpp

```diff
@@ -551,6 +551,29 @@ namespace PocketDb
         });
     }
 
+    // Remove all transactions and PocketNet payloads which are older than the specified time
+    // and have not yet been added to a block 
+    void TransactionRepository::CleanExpiredTransactions(int64_t time)
+    {
+        TryTransactionStep(__func__, [&]()
+        {
+            std::vector<std::string> hashes;
+
+            // Get list of expired transaction hashes which have not been added to a block
+            auto stmt = SetupSqlStatement(R"sql(
+                select Hash, Time from Transactions t where t.Hash isnull and t.Height isnull and t.Time < ?
+            )sql");
+            TryBindStatementInt64(stmt, 1, time);
+            while (sqlite3_step(*stmt) == SQLITE_ROW)
+                hashes.push_back( std::string(reinterpret_cast<const char*>(sqlite3_column_text(*stmt, 0))) );
+
+            for (std::string hash : hashes)
+            {
+                CleanTransaction(hash);
```

Comment:
`CleanTransaction` uses a transaction inside - this will cause problems in the database due to the nesting of transactions

Reply:
Do you mean that there might be dependent transactions in the mempool that depend on the output of the transaction being deleted?  I can delete this function either way because I never saw it being called in my testing.

Reply:
Oh, no, I'm talking about SQLite Transaction Begin/Commit
TryTransactionStep creates a database "transaction", and there will be problems when embedding here.

Reply:
Take out the `std::array::vector<std::string> hashes;` and loop with CleanTransaction outside TryTransactionStep

Reply:
@tawmaz

Reply:
Got it, working on testing this change now on 0.21 branch